ui/src/app/cohort-search/redux/reducer.ts

```diff
@@ -510,26 +512,49 @@ export const rootReducer: Reducer<CohortSearchState> =
       case RESET_STORE:
         return initialState;
 
-    /**
-     * Cohort Review Charts
-     */
+      /**
+       * Cohort Review Charts
+       */
       case BEGIN_CHART_DATA_REQUEST:
-          return state
-              .deleteIn(
-                  ['reviewChartData', 'request', action.domain])
-              .setIn(
-                  ['reviewChartData', 'request', action.ns,
-                      action.wsid, action.cid, action.cdrid, action.domain, action.limit],
-                  true);
-
-        case LOAD_CHART_RESULTS:
-          return state
-              .setIn(
-                  ['reviewChartData', 'domainCharts', action.domain]
-                  , fromJS(action.results ))
-              .deleteIn(
-                  ['reviewChartData', 'request', action.ns,
-                    action.wsid, action.cid, action.cdrid, action.domain, action.limit]);
+        return state
+          .deleteIn(
+            ['reviewChartData', 'request', action.cid, action.domain])
+          .setIn(
+            ['reviewChartData', 'request', action.ns,
+              action.wsid, action.cid, action.cdrid, action.domain, action.limit],
+            true);
+
+      case LOAD_CHART_RESULTS:
+        return state
+          .setIn(
+            ['reviewChartData', 'domainCharts', action.cid, action.domain]
+            , fromJS(action.results))
+          .deleteIn(
+            ['reviewChartData', 'request', action.ns,
+              action.wsid, action.cid, action.cdrid, action.domain, action.limit]);
+      /**
+       * Cohort Individual Participants Charts
+       */
+
+      case BEGIN_INDIVIDUAL_PARTICIPANTS_CHART_REQUEST:
+        return state
+          .deleteIn(
+            ['individualChartData', 'request', action.cid , action.participantsId, action.domain])
+          .setIn(
+            ['individualChartData', 'request', action.ns,
+              action.cid, action.participantsId, action.domain,
+              action.wsid,  action.cdrid,  action.limit],
+            true);
```

Comment:
We didnt had any requirement, so I though as three charts are loading almost in same time I just kept one spinner , Not sure how it will look for three spinner, i will do changes and will show you.